### ISSUE TITLE
Improve error reporting: MissingMethodException

### ISSUE THREAD

**Issue by vilinski** [Microsoft/visualfsharp#1119]
Related to https://github.com/Microsoft/visualfsharp/issues/1103

The reproducing steps or cause might be fully incorrect, but this is one of the most annoying things during working with F# Projects in different environments.

* Create a Solution
* target it to FSharp.Core 4.3.0.0
* add some References, e.g. Unquote, NUnit (not sure which pattern is here?)
* Write some tests

The test execution fails with an Exception something like that:

```
   System.MissingMethodException : Method not found: "Void Swensen.Unquote.UnquotedExpression..ctor(Microsoft.FSharp.Quotations.FSharpExpr, Microsoft.FSharp.Core.FSharpOption`1<Microsoft.FSharp.Collections.FSharpMap`2<Microsoft.FSharp.Quotations.FSharpVar,System.Object>>)".
   by MyProject.TestNamespace.TestName()
```

The message is less helpful, because all methods are actually on their places. It is just wrong FSharp.Core.dll referenced
The workaround to this is just create an `App.config` file to the (test) project with following content:

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <Paket>True</Paket>
        <assemblyIdentity name="FSharp.Core" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-999.999.999.999" newVersion="4.4.0.0" />
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
</configuration>
```

This redirects any version of FSharp.Core to the most available specified one - 4.4.0.0.

May be here is a better workaround too.

**Comment by forki** [comment]
I *assume* we can't fix this in the compiler. The compilation phase is already done here. This is most probably a runtime issue and should be raised in the framework repos

**Comment by isaacabraham** [comment]
To me this is a runtime issue, not compiler issue - I think it fails outside the scope of this piece of work.

**Comment by dsyme** [comment]
@vilinski  We really need _exact_ repro steps for instances this problem please :) The ISSUE_TEMPLATE.md has sections that need to be filled in before we can even investigate an issue - please reopen the issue with exact reproduction steps, thanks.

The problem is almost invariably a problem with a specific F# project template or nuget package missing some setting, and we need to know exactly the circumstances that cause the problem.
